src/bgc_md2/models/CARDAMOM/source.py

```diff
@@ -34,36 +39,90 @@ def create_pwc_model_run_fd(ens, lat, lon):
+    'gpp_to_root':              'Fraction of GPP allocated to root carbon pool',
+    'gpp_to_wood':              'Fraction of GPP allocated to wood carbon pool',
+    'fire_em_labile':           'external flux rate of labile carbon by fire',
+    'fire_em_foliar':           'external flux rate for leaf pool by fire',
+    'fire_em_root':             'external flux rate for root pool by fire',
+    'fire_em_wood':             'external flux rate for wood pool by fire',
+    'fire_em_litter':           'external flux rate for litter pool by fire',
+    'fire_em_som':              'external flux rate for soil pool by fire',
+    'hetresp_litter':           'heterotrophic respiration rate of litter pool',
+    'hetresp_som':              'heterotrophic respiration rate of som pool',
+    'labile_to_foliar':         'internal flux rate',
+    'fire_labile_to_litter':    'internal flux rate',
+    'fire_foliar_to_litter':    'internal flux rate',
+    'fire_wood_to_som':         'internal flux rate',
+    'wood_to_soilc':            'internal flux rate',
+    'fire_root_to_litter':      'internal flux rate',
+    'root_to_litter':           'internal flux rate',
+    'fire_litter_to_som':       'internal flux rate',
+    'litter_to_som':            'internal flux rate',
+}
+
+for name in sym_dict.keys():
+    var(name)
+
+
 t = TimeSymbol('t')
 ms = load_model_structure()
-x = StateVariableTuple(ms.pool_names)
-def make_input_flux(sv_name):
-    sv = Symbol(sv_name)
-    fluxrate_densities = [ Symbol(name) for name in ms.external_input_structure]
-    if sv in fluxrate_densities:
-        fl = sum(fluxrate_densities)*sv
-    else:
-        fl = 0
-    return fl
-u = InputTuple((make_input_flux(key) for key in ms.pool_names))
+x = StateVariableTuple(tuple(Symbol(name) for name in ms.pool_names))
+gpp = Symbol('gpp')
+
+
+def make_internal_flux(sv_name_tup):
+    source, target = map(Symbol, sv_name_tup)
+    fluxrate_densities = [
+        Symbol(var_name)
+        for var_name in ms.horizontal_structure[sv_name_tup]
+    ]
+    return sum(fluxrate_densities) * source
+
+
 mvs = MVarSet({
-    #BibInfo(# Bibliographical Information
-    #    name="CARDAMOM",
-    #    longName="to be ", 
-    #    version="2.6",
-    #    entryAuthor="",
-    #    entryAuthorOrcid="",
-    #    entryCreationDate="22/3/2016",
-    #    doi="",
-    #    #further_references=BibInfo(doi="10.5194/bg-10-2255-2013"),
-    #    sym_dict=sym_dict
-    #),
-    #A,  # the overall compartmental matrix
-    u,  # imput tuple
-    t,  # time symbol 
+    BibInfo(# Bibliographical Information
+       name="CARDAMOM",
+       longName="?",
+       version="?",
+       entryAuthor="Holger Metzler",
+       entryAuthorOrcid="",
+       entryCreationDate="02/02/2021",
+       doi="",
+       #further_references=BibInfo(doi="10.5194/bg-10-2255-2013"),
+       sym_dict=sym_dict
+    ),
+    InFluxesBySymbol(
+        {
+            Symbol(name): gpp * Symbol(val[0])
```

Comment:
The symbol should be named "npp" instead of "gpp". Even though the data variable in the CARDAMOM netCDF file is named "gpp", there is not autotrophic respiration in the model, hence it is in fact "npp".